### DISCUSSION TITLE
Updated Examples

### DISCUSSION THREAD

**Discussion by RMagician** [denisneuf/python-amazon-ad-api#137] [Announcements]
**Describe the bug**
Hey, first of all thanks for the api wrapper. I'm trying to use it for v3 Reports, but not being able to. If you could provide some updated examples would be great. The example in the readme is for v2 and deprecated.

**To Reproduce**
```
from ad_api.api.reports import Reports

from config import AmazonAdAPI

my_credentials = AmazonAdAPI.get_credentials()



params = {
        "name":"",
        "startDate":"2023-04-01",
        "endDate":"2023-04-08",
        "configuration":{
            "adProduct":"SPONSORED_PRODUCTS",
            "groupBy":["campaign","adGroup"],
            "columns":["adGroupId","campaignId", "impressions", "clicks", "cost", "purchases1d", "purchases7d", "purchases14d", "purchases30d","date"],
            "reportTypeId":"spCampaigns",
            "format":"GZIP_JSON"
        }
    }

result=Reports(credentials=my_credentials).post_report(**params)
```

**Expected behavior**
I've tried a few different scenarios but keep getting stuck on different errors.

**Desktop (please complete the following information):**
 Python 3.11.1
 python-amazon-ad-api 0.4.2

**Comment by denisneuf** [comment]
You don't need to unpack your dictionary:

result=Reports(credentials=my_credentials).post_report(****params**)

*Suggestion use try catch to get some Exception if happen + Debug while testing.

You need to pass a body json type value, anyway take my example I used to test the new version 3 of that endpoint:

```
import logging
from ad_api.api import Reports
from ad_api.base import AdvertisingApiException
import json

logging.basicConfig(
    level=logging.INFO,
    format="%(asctime)s:%(levelname)s:%(message)s"
)



def download_report(url):

    download_type = "data"

    result = Reports(debug=True).download_report(
        url=url,
        format=download_type
    )

    payload = result.payload

    logging.info(payload)


def get_report(report_id):

    try:

        result = Reports(debug=True).get_report(
            reportId=report_id
        )

    except AdvertisingApiException as error:
        logging.error(error)

    payload = result.payload

    logging.info(payload)


def post_report(request):

    logging.info("---------------------------------")
    logging.info("Offline Report")
    logging.info("---------------------------------")


    try:


        result = Reports(debug=True).post_report(
            body=json.dumps(request)
        )

        payload = result.payload

        logging.info(payload)


    except AdvertisingApiException as error:
        logging.error(error)






if __name__ == '__main__':


    request = \
        {
            "startDate": "2023-01-09",
            "endDate": "2023-01-09",
            "configuration": {
                "adProduct": "SPONSORED_PRODUCTS",
                "groupBy": ["advertiser"],
                "columns": ["impressions", "clicks", "cost", "campaignStatus", "advertisedAsin", "date"],
                "reportTypeId": "spAdvertisedProduct",
                "timeUnit": "DAILY",
                "format": "GZIP_JSON"
            }
        }

    post_report(request)

    # report_id = 'aa774e40-2634-4740-b9a2-123456789'
    # get_report(report_id)

    # url = 'https://offline-report-storage-eu-west-1-prod.s3.eu-west-1.amazonaws.com/******'
    # download_report(url)
```

1. First you post_report and you will get a report_id
2. Second get_report(report_id) whith the report id you obtain in step 1
3. Finally download the report with the url you obtain in step2

Let me know if it works for you.

Kindest regards,
Daniel

**Reply by RMagician** [reply]
Thank you, very helpful. 

I get service unavailable when running the code. Is that an issue on my side ? 


`2023-04-14 12:25:36,314:INFO:{'_content': b'<html><body><b>Http/1.1 Service Unavailable</b></body> </html>',`

I've set env to production, but still keeps going to 
`2023-04-14 12:25:36,311:INFO:POST https://advertising-api-test.amazon.com/reporting/reports`

could that be why ? 

edit - indeed the issue seems to be that it is going to the `advertising-api-test` endpoint rather than `advertising-api-eu` in my case

edit 2 - Env was incorrectly set up, I've now downloaded a report, thank you :)

**Comment by denisneuf** [comment]
Sure it is, according the log you are posting to sandbox endpoint that as far as I know this new endpoint do not support.

Check that your .env file in the same level that your code is set to:

# environment variables defined inside a .env file
AWS_ENV=PRODUCTION
# AWS_ENV=SANDBOX
